### ISSUE TITLE
How to package templates in a cargo library but not the binary?

### ISSUE THREAD

**Issue by schneems** [Keats/tera#788]
First off, thanks for tera. It's really nice. I'm interested in packaging templates into a crate for publishing, but not having them in the resulting binary. I see https://github.com/Keats/tera/issues/372 and I've asked on stack overflow https://stackoverflow.com/questions/74606543/ship-tera-templates-in-a-crate-without-embedding-them-into-the-binary?noredirect=1#comment131691321_74606543 but I've not found if it's possible to reference assets.

Essentially I want it to work in the same way that `rails new` does where the templates ship in the library on disk and aren't loaded into memory unless a `rails new` command is run, then they're read from disk and used as templates.

I'm curious if anyone in the project (or the community) has found a way to reference the install location of a crate to achieve similar functionality.

**Comment by Keats** [comment]
I don't think you can ship assets without including them in the binary? I've made https://github.com/Keats/kickstart as a scaffolding tool but it still requires you to provide the templates externally. 
What's the issue with including them in the binary?

**Comment by Wicpar** [comment]
i think it would be easier to just use consts with `include_str("...")` and then include the templates with [add_raw_templates](https://docs.rs/tera/latest/tera/struct.Tera.html#method.add_raw_templates)

if you don't call that function the const should not include within the binary. If you need to compile a template without the source, build.rs or a proc macro is the easiest solution.

**Comment by Keats** [comment]
Closing as part of the Tera v2 release. If this is still relevant in v2, please reopen — see the 
  migration guide: https://github.com/Keats/tera/blob/master/MIGRATION.md
